### ISSUE TITLE
Useless query being generated when including relations + buggy behaviour in conjunction with remote models 

### ISSUE THREAD

**Issue by horiaradu** [strongloop/loopback-datasource-juggler#1006]
In [include.js](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/include.js), in various places, the following code occurs:

```
...
filter.where[modelToIdName] = {
  inq: uniq(targetIds),
};
...
findWithForeignKeysByPage(relation.modelTo, filter, modelToIdName, 0, options, targetsFetchHandler);
```

The problem is that, if ```uniq(targetIds)``` is an empty array, the query is useless since it will not return any results. If the model is a remote model and thus the query requires a remote call, this call becomes expensive since you pay the price of a network request. 

This query can't be controlled by the user since it is being generated by including relationships:
```
Model.find({where: {...}, include: ['relatedModel'])
```

This alone wouldn't be an issue, but taken in conjunction with [this](https://github.com/strongloop/strong-remoting/issues/324), it becomes a problem. 

In short: when using a remote model, if ```targetIds``` array is empty, the where clause disappears from the remote request URL and, instead of getting back an empty array, you will get back **all** the related items.

In conjunction with the fix in strong-remoting, I would also not perform the query when the array is empty.
